### PR TITLE
Made 'notes' field optional

### LINKED ISSUE #5: Make 'notes' subfield optional
According to @pauladkisson, the 'notes' field of the FibersTable is required

Relevant section of schema: https://github.com/catalystneuro/ndx-photometry/blob/master/spec/ndx-photometry.extensions.yaml#L30-L35

This type of field is usually regarded as optional by NWB standards (such at the NWB file metadata level)

We are also dealing with a conversion where no extra information needs to be included here, so we'd rather not have to set the property

### PR DESCRIPTION
**Pull request by pauladkisson**
Fixes #5

### CODE REVIEW

**Comment by pauladkisson** [comment]
I noticed during this fix that there are 2 (potentially diverging) different `spec` folders (`ndx-photometry/spec`, `ndx-photometry/src/pynwb/spec`) each with their own `ndx-photometry.extensions.yaml` and `ndx-photometry.namespace.yaml` files.  For this fix, I only changed `src/spec/create_extension_spec.py`, which then only altered the `ndx-photometry/spec/ndx-photometry.extensions.yaml` file (NOT the `ndx-photometry/src/pynwb/spec/ndx-photometry.extensions.yaml` file).

I would expect that these files should always be identical, but I don't know who uses the `ndx-photometry/spec` versions (or why they exist at all).

**Comment by CodyCBakerPhD** [comment]
> I noticed during this fix that there are 2 (potentially diverging) different spec folders (ndx-photometry/spec, ndx-photometry/src/pynwb/spec) each with their own ndx-photometry.extensions.yaml and ndx-photometry.namespace.yaml files.

Can you link me to the diverging spec folder under `ndx-photometry/src/pynwb/spec`? I can't see it on my end https://github.com/catalystneuro/ndx-photometry/tree/master/src/pynwb/ndx_photometry

**Comment by pauladkisson** [comment]
> Can you link me to the diverging spec folder under `ndx-photometry/src/pynwb/spec`? I can't see it on my end https://github.com/catalystneuro/ndx-photometry/tree/master/src/pynwb/ndx_photometry

Actually, it's in the [.gitignore](https://github.com/pauladkisson/ndx-photometry/blob/6e3de1a6bb3e3664f7deeed99e3704a1db51aa96/.gitignore#LL7C1-L8C37), so you can't see it on github...

**Comment by CodyCBakerPhD** [comment]
The `.gitignore` just means it wouldn't be added automatically by `git add *`-like operations; it's also in the [core template](https://github.com/nwb-extensions/ndx-template/blob/main/%7B%7B%20cookiecutter.namespace%20%7D%7D/.gitignore#L7-L8), which is how it ended up here

Anyway, can you add a test that it's possible to make the FibersTable without notes? Then this should be good to go

**Comment by pauladkisson** [comment]
Ok, so what's happening here is that when ndx-photometry is *installed* (`pip install -e .`) the extra `ndx-photometry/src/pynwb/spec` folder gets created with the appropriate .yaml files.

But, if you edit the extension spec via `src/spec/create_extension_spec.py` (and then run `python3 src/spec/create_extension_spec.py`), it only changes the `ndx-photometry/spec/ndx-photometry.extensions.yaml` file.

As a consequence, these changes only take effect after *reinstalling* ndx-photometry.

I can think of a few possible fixes:
1) Just update the readme with a 'Development' section that makes note of this behavior for any future dev who wants to update the extension spec
2) Change the `ndx-photometry/src/pynwb/ndx_photometry/__init__.py` file so that it always references the version-controlled spec (`ndx-photometry/spec`)
3) Change the `ndx-photometry/src/spec/create_extension_spec.py` file so that it updates both specs together

**Comment by pauladkisson** [comment]
> Anyway, can you add a test that it's possible to make the FibersTable without notes? Then this should be good to go

Done!

**Comment by CodyCBakerPhD** [comment]
> Ok, so what's happening here is that when ndx-photometry is installed ...

@pauladkisson OK, thanks for digging into this

attn: @rly can you shed light on this behavior? Is it expected? Should we raise an issue on ndx-template?

File: src/pynwb/tests/integration/test_photometry.py [hunk 1]
```diff
@@ -136,7 +136,6 @@ def test_roundtrip(self):
             photodetector=0,
             fluorophores=[0],
             location='my location',
-            notes='notes'
```
Comment (CodyCBakerPhD) [inline src/pynwb/tests/integration/test_photometry.py:139]: Strictly speaking there should be 

1) a test that you _can_ write the `notes` field successfully and it ends up in the neurodata type or output NWB file

2) a test that you can skip the field and it doesn't become populated in the resulting neurodata type in memory or written to the output NWB file

Right now there only seems to be the latter

Comment (pauladkisson) [reply src/pynwb/tests/integration/test_photometry.py:139]: None of the other optional fields (fiber model number, dichroic_model_number, etc.) are tested in the way you described.  And all the tests in the unit folder are commented out for some reason.  So short of a full overhaul of the testing suite (which seems way beyond the scope of this fix), I think just omitting 'notes' is the way to do it.

Comment (CodyCBakerPhD) [reply src/pynwb/tests/integration/test_photometry.py:139]: > None of the other optional fields (fiber model number, dichroic_model_number, etc.) are tested in the way you described. And all the tests in the unit folder are commented out for some reason. So short of a full overhaul of the testing suite (which seems way beyond the scope of this fix), I think just omitting 'notes' is the way to do it.

None of those underwent a change after initial creation, and this is also our first time giving this attention

A simple unit test should be quick and easy to put together, and never hurts anything

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by rly** [comment]
> attn: @rly can you shed light on this behavior? Is it expected? Should we raise an issue on ndx-template?

Just following up belatedly that this behavior of having two spec folders was expected and a result of how the spec needed to be copied into a different folder in order to be packaged. I have reworked ndx-template so that it no longer has to do that, because I agree, it was confusing!

**Pull request merged**